lib/internal/modules/esm/formats.js

```diff
@@ -35,7 +38,27 @@ function mimeToFormat(mime) {
   return null;
 }
 
+function guessExtensionlessModule(url) {
+  if (!experimentalWasmModules)
+    return 'module';
+
+  const magic = new Uint8Array(4);
+  let fd;
+  try {
+    fd = openSync(url);
+    readSync(fd, magic);
+    if (magic[0] === 0x00 && magic[1] === 0x61 && magic[2] === 0x73 && magic[3] === 0x6d) {
+      return 'wasm';
+    }
```

Comment:
This happens only for extensionless files, right? If that's the case, that seems absolutely OK as a tradeoff, the workaround is to use the `.mjs` extension.

Reply:
That's correct, the check is only for local extensionless files in `module` scope.

Reply:
Why do we need a tradeoff? Can’t we keep the file contents in memory once we have them, so that later on when the `load` hook runs it uses the contents from memory rather than refetching them from disk?